src/algebra/ordered_group.lean

```diff
@@ -646,129 +802,355 @@ lemma with_top.add_lt_add_iff_right
+@[to_additive]
+lemma le_mul_of_inv_mul_le (h : b⁻¹ * a ≤ c) : a ≤ b * c :=
+begin
+  have h := mul_le_mul_left' h,
+  rwa mul_inv_cancel_left at h
+end
 
-lemma lt_of_neg_lt_neg (h : -b < -a) : a < b :=
-neg_neg a ▸ neg_neg b ▸ neg_lt_neg h
+@[to_additive]
+lemma inv_mul_le_of_le_mul (h : a ≤ b * c) : b⁻¹ * a ≤ c :=
+begin
+  have h := mul_le_mul_left' h,
+  rwa inv_mul_cancel_left at h
+end
 
-lemma pos_of_neg_neg (h : -a < 0) : 0 < a :=
-have -a < -0, by rwa neg_zero,
-lt_of_neg_lt_neg this
+@[to_additive]
+lemma le_mul_of_inv_mul_le_left (h : b⁻¹ * a ≤ c) : a ≤ b * c :=
+le_mul_of_inv_mul_le h
 
-lemma neg_neg_of_pos (h : 0 < a) : -a < 0 :=
-have -a < -0, from neg_lt_neg h,
-by rwa neg_zero at this
+@[to_additive]
+lemma inv_mul_le_left_of_le_mul (h : a ≤ b * c) : b⁻¹ * a ≤ c :=
+inv_mul_le_of_le_mul h
 
-lemma neg_of_neg_pos (h : 0 < -a) : a < 0 :=
-have -0 < -a, by rwa neg_zero,
-lt_of_neg_lt_neg this
+@[to_additive]
+lemma le_mul_of_inv_mul_le_right (h : c⁻¹ * a ≤ b) : a ≤ b * c :=
+by { rw mul_comm, exact le_mul_of_inv_mul_le h }
 
-lemma neg_pos_of_neg (h : a < 0) : 0 < -a :=
-have -0 < -a, from neg_lt_neg h,
-by rwa neg_zero at this
+@[to_additive]
+lemma inv_mul_le_right_of_le_mul (h : a ≤ b * c) : c⁻¹ * a ≤ b :=
+by { rw mul_comm at h, apply inv_mul_le_left_of_le_mul h }
 
-lemma le_neg_of_le_neg (h : a ≤ -b) : b ≤ -a :=
+@[to_additive]
+lemma mul_lt_of_lt_inv_mul (h : b < a⁻¹ * c) : a * b < c :=
 begin
-  have h := neg_le_neg h,
-  rwa neg_neg at h
+  have h := mul_lt_mul_left' h a,
+  rwa mul_inv_cancel_left at h
 end
 
-lemma neg_le_of_neg_le (h : -a ≤ b) : -b ≤ a :=
+@[to_additive]
+lemma lt_inv_mul_of_mul_lt (h : a * b < c) : b < a⁻¹ * c :=
 begin
-  have h := neg_le_neg h,
-  rwa neg_neg at h
+  have h := mul_lt_mul_left' h (a⁻¹),
+  rwa inv_mul_cancel_left at h
 end
 
-lemma lt_neg_of_lt_neg (h : a < -b) : b < -a :=
+@[to_additive]
+lemma lt_mul_of_inv_mul_lt (h : b⁻¹ * a < c) : a < b * c :=
 begin
-  have h := neg_lt_neg h,
-  rwa neg_neg at h
+  have h := mul_lt_mul_left' h b,
+  rwa mul_inv_cancel_left at h
 end
 
-lemma neg_lt_of_neg_lt (h : -a < b) : -b < a :=
+@[to_additive]
+lemma inv_mul_lt_of_lt_mul (h : a < b * c) : b⁻¹ * a < c :=
 begin
-  have h := neg_lt_neg h,
-  rwa neg_neg at h
+  have h := mul_lt_mul_left' h (b⁻¹),
+  rwa inv_mul_cancel_left at h
 end
 
+@[to_additive]
+lemma lt_mul_of_inv_mul_lt_left (h : b⁻¹ * a < c) : a < b * c :=
+lt_mul_of_inv_mul_lt h
+
+@[to_additive]
+lemma inv_mul_lt_left_of_lt_mul (h : a < b * c) : b⁻¹ * a < c :=
+inv_mul_lt_of_lt_mul h
+
+@[to_additive]
+lemma lt_mul_of_inv_mul_lt_right (h : c⁻¹ * a < b) : a < b * c :=
+by { rw mul_comm, exact lt_mul_of_inv_mul_lt h }
+
+@[to_additive]
+lemma inv_mul_lt_right_of_lt_mul (h : a < b * c) : c⁻¹ * a < b :=
+by { rw mul_comm at h, exact inv_mul_lt_of_lt_mul h }
+
+@[to_additive]
+lemma mul_le_mul_three {a b c d e f : α} (h₁ : a ≤ d) (h₂ : b ≤ e) (h₃ : c ≤ f) :
+      a * b * c ≤ d * e * f :=
```

Comment:
I thought they liked
```
le_trans (mul_le_mul'' (mul_le_mul'' h₁ h₂) h₃) $ le_refl _
```
better?

Reply:
wait -- whatever is le_trans even doing there? `mul_le_mul'' (mul_le_mul'' h₁ h₂) h₃`?